objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/neuralNet/YoloProcessor.java

```diff
@@ -230,17 +234,41 @@ private void setup(Map<String, ?> neuralNet, String sourceName, String modelDire
    }
 
     /**
-     * Run the image through a YOLO net. May save the resulting image depending on the settings.
+     * Deprecated method, should not be used anymore.
      */
     @Override
     public NeuralNetResults processImage(byte[] image) throws ImageProcessingException {
+        log.debug("Deprecated method, should no longer be used.");
+        return processImage(null, image);
+    }
+    
+    /**
+     * Deprecated method, should not be used anymore.
+     */
+    @Override
+    public NeuralNetResults processImage(byte[] image,  Map<String, ?> request) throws ImageProcessingException {
+        log.debug("Deprecated method, should no longer be used.");
+        return processImage(null, image, request);
+    }
+    
+    /**
+     * Run the image through a YOLO net. May save the resulting image depending on the settings.
+     */
+    @Override
+    public NeuralNetResults processImage(Map<String, ?> processingParams, byte[] image) throws ImageProcessingException {
+        Date timestamp;
+        if (processingParams != null) {
+            timestamp = (Date) processingParams.get("timestamp");
```

Comment:
This doesn't verify that `IMAGE_TIMESTAMP` exists -- only that some `Map` was passed in.  Either change the test to include that test, or handle its being null.  I don't think you want/expect `timestamp` to be null.